src/lib/util/options.h

```diff
@@ -152,15 +153,15 @@ class core_options
 
 	// getters
 	const std::string &command() const { return m_command; }
-	const entry *get_entry(const std::string &name) const;
-	entry *get_entry(const std::string &name);
-	std::vector<entry::ptr> &entries() { return m_entries; }
-	const std::vector<entry::ptr> &entries() const { return m_entries; }
+	const entry::shared_ptr get_entry(const std::string &name) const;
+	entry::shared_ptr get_entry(const std::string &name);
+	std::vector<entry::shared_ptr> &entries() { return m_entries; }
```

Comment:
Is it really safe to expose a non-const reference to the entries like this?  You can't enforce any kind of semantics on it.  Since it's an array of pointers, the entries themselves can still be modified with the const accessor on the next line.